Mathlib/Analysis/InnerProductSpace/ProdL2.lean

```diff
@@ -0,0 +1,44 @@
+/-
+Copyright (c) 2023 Moritz Doll. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Moritz Doll
+-/
+
+import Mathlib.Analysis.InnerProductSpace.Projection
+import Mathlib.Analysis.NormedSpace.ProdLp
+
+/-!
+# `L²` inner product space structure on products of inner product spaces
+
+The `L²` norm on product of two inner product spaces is compatible with an inner product
+$$
+\langle x, y\rangle = \langle x_1, y_1 \rangle + \langle x_2, y_2 \rangle.
+$$
+This is recorded in this file as an inner product space instance on `ProdLp 2`.
```

Comment:
I think that all references to `ProdLp` should be replaced with references to `WithLp`, here and below.